cmd/build.go

```diff
@@ -183,3 +237,65 @@ func findDockerRootDir(workingDir, buildPath string) (string, error) {
 		searchPath = dir
 	}
 }
+
+func createDockerExecutor(shell exec.Executor, contextRootDir string) *kaniko.DockerExecutor {
+	dockerCfg := kaniko.ContainerConfig{
+		Image: "gcr.io/kaniko-project/executor:v1.7.0",
+		Env:   map[string]string{},
+		Volumes: map[string]string{
+			contextRootDir: contextRootDir,
+		},
+	}
+
+	return kaniko.NewDockerExecutor(shell, dockerCfg)
+}
+
+func createKubernetesExecutor() (*kaniko.KubernetesExecutor, error) {
+	k8sClient, err := kube.New("")
+	if err != nil {
+		return nil, fmt.Errorf("could not get kube client from context: %w", err)
+	}
+
+	executor := kaniko.NewKubernetesExecutor(k8sClient.ClientSet, kaniko.JobConfig{
```

Comment:
Many things in this function are private and static :)

Reply:
Yes, every hardcoded setting here are planned to be moved to a config file when we support them